pygmt/datasets/earth_relief.py

```diff
@@ -103,12 +119,17 @@ def load_earth_relief(resolution="01d", region=None, registration=None):
             f"resolution '{resolution}' is not supported."
         )
 
+    # Choose earth relief data prefix
+    earth_relief_prefix = "earth_relief_"
+    if use_srtm and resolution in land_only_srtm_resolutions:
+        earth_relief_prefix = "srtm_relief_"
+
     # different ways to load tiled and non-tiled earth relief data
     # Known issue: tiled grids don't support slice operation
     # See https://github.com/GenericMappingTools/pygmt/issues/524
     if region is None:
         if resolution in non_tiled_resolutions:
-            fname = which(f"@earth_relief_{resolution}{reg}", download="a")
+            fname = which(f"@{earth_relief_prefix}{resolution}{reg}", download="a")
             with xr.open_dataarray(fname) as dataarray:
                 grid = dataarray.load()
                 _ = grid.gmt  # load GMTDataArray accessor information
```

Comment:
Lines 131 to 141 can be refactored as follows to remove the `else` statement (Sorry I can't make a suggested change because it goes over some unchanged lines):

```python
        if resolution not in non_tiled_resolutions:
            raise GMTInvalidInput(
                f"'region' is required for Earth relief resolution '{resolution}'."
            )
        fname = which(f"@{earth_relief_prefix}{resolution}{reg}", download="a")
        with xr.open_dataarray(fname) as dataarray:
            grid = dataarray.load()
            _ = grid.gmt  # load GMTDataArray accessor information
```
This is actually an idea from https://github.com/weiji14/pygmt/pull/276/files#r586818072, visual diff below:

```diff
diff --git a/pygmt/datasets/earth_relief.py b/pygmt/datasets/earth_relief.py
index d57948e95..9e0feb27e 100644
--- a/pygmt/datasets/earth_relief.py
+++ b/pygmt/datasets/earth_relief.py
@@ -107,15 +107,14 @@ def load_earth_relief(resolution="01d", region=None, registration=None):
     # Known issue: tiled grids don't support slice operation
     # See https://github.com/GenericMappingTools/pygmt/issues/524
     if region is None:
-        if resolution in non_tiled_resolutions:
-            fname = which(f"@earth_relief_{resolution}{reg}", download="a")
-            with xr.open_dataarray(fname) as dataarray:
-                grid = dataarray.load()
-                _ = grid.gmt  # load GMTDataArray accessor information
-        else:
+        if resolution not in non_tiled_resolutions:
             raise GMTInvalidInput(
                 f"'region' is required for Earth relief resolution '{resolution}'."
             )
+        fname = which(f"@earth_relief_{resolution}{reg}", download="a")
+        with xr.open_dataarray(fname) as dataarray:
+            grid = dataarray.load()
+            _ = grid.gmt  # load GMTDataArray accessor information
     else:
         grid = grdcut(f"@earth_relief_{resolution}{reg}", region=region)
```

Reply:
> This is actually an idea from [weiji14/pygmt#276 (files)](https://github.com/weiji14/pygmt/pull/276/files#r586818072),

[Sourcery](https://sourcery.ai/) looks like an amazing tool!

Reply:
@willschlitzer done in a8ac453. Amazing tool.